x-pack/plugins/endpoint/server/routes/alerts.ts

```diff
@@ -5,27 +5,101 @@
  */
 
 import { IRouter } from 'kibana/server';
+import { RequestHandler } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema, TypeOf } from '@kbn/config-schema';
 
-import json from './sampledata.json';
+import {
+  getPagingProperties,
+  kibanaRequestToAlertListQuery,
+} from '../services/endpoint/alert_query_builders';
+
+import { AlertData, AlertResultList } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const ALERTS_ROUTE = '/api/endpoint/alerts';
+
+export const reqSchema = schema.object({
+  page_size: schema.number({ defaultValue: 10, min: 1, max: 10000 }),
+  page_index: schema.number({ defaultValue: 0, min: 0 }),
+});
+
+export function registerAlertRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const alertsHandler: RequestHandler<unknown, TypeOf<typeof reqSchema>> = async (
+    ctx,
+    req,
+    res
+  ) => {
+    try {
+      const queryParams = await getPagingProperties(req, endpointAppContext);
+      const reqBody = await kibanaRequestToAlertListQuery(queryParams, endpointAppContext);
+      const response = (await ctx.core.elasticsearch.dataClient.callAsCurrentUser(
+        'search',
+        reqBody
+      )) as SearchResponse<AlertData>;
+      return res.ok({ body: mapToAlertResultList(endpointAppContext, queryParams, response) });
+    } catch (err) {
+      return res.internalError({ body: err });
+    }
+  };
 
-export function registerAlertRoutes(router: IRouter) {
   router.get(
     {
-      path: '/api/endpoint/alerts',
-      validate: false,
+      path: ALERTS_ROUTE,
+      validate: {
+        query: reqSchema,
+      },
       options: { authRequired: true },
     },
-    async (context, req, res) => {
-      try {
-        return res.ok({
-          body: json,
-          headers: {
-            'Content-Type': 'application/json',
-          },
-        });
-      } catch (err) {
-        return res.internalError({ body: err });
-      }
-    }
+    alertsHandler
   );
+
+  router.post(
+    {
+      path: ALERTS_ROUTE,
+      validate: {
+        body: reqSchema,
+      },
+      options: { authRequired: true },
+    },
+    alertsHandler
+  );
+}
+
+function mapToAlertResultList(
+  endpointAppContext: EndpointAppContext,
+  queryParams: Record<string, any>,
+  searchResponse: SearchResponse<AlertData>
+): AlertResultList {
+  interface Total {
+    value: number;
+    relation: string;
+  }
+
+  let totalNumberOfAlerts: number = 0;
+  let totalIsLowerBound: boolean = false;
+
+  // We handle 2 separate schemas for the response below, due to: https://github.com/elastic/kibana/issues/56694
+  if (typeof searchResponse?.hits?.total === 'object') {
```

Comment:
nit: I think we're only supporting Kibana 7.x and onward right? So we could probably assume that total will always be an object?

Reply:
We need to do the type check anyway to prevent type-check errors due to the outdated schema, so we have to handle the failure case one way or another. I figured it would be better to just handle it rather than return an error, since it IS something we can handle... happy to discuss if you feel strongly about this.